### PR TITLE
Add Default Interface Implementation tests for VB

### PR DESCRIPTION
**Pull request by AlekseyTs**
Related to #35820, #35821, #35823, #35824, #35827, #35834, #35852, #35885, #35911, #35948.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, this is a test only change

File: src/Compilers/VisualBasic/Test/Symbol/Microsoft.CodeAnalysis.VisualBasic.Symbol.UnitTests.vbproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Library</OutputType>
-    <TargetFrameworks>$(RoslynPortableTargetFrameworks)</TargetFrameworks>
+    <TargetFrameworks>net472;netcoreapp3.0</TargetFrameworks>
```
Comment (jcouv) [inline src/Compilers/VisualBasic/Test/Symbol/Microsoft.CodeAnalysis.VisualBasic.Symbol.UnitTests.vbproj]: >net472;netcoreapp3.0 [](start = 22, length = 20)

What values would `$(RoslynPortableTargetFrameworks)` produce? (ie. what's the difference?)

File: src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs [hunk 2]
```diff
@@ -39855,6 +39855,94 @@ static void Main()
             }
         }
 
+        [Fact]
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: > [Fact] [](start = 7, length = 7)

Not related to this PR: have we thought about the use of Obsolete attribute in DIM? Consider adding some tests if interesting.

**Comment by jcouv** [comment]
>         Private Function GetCSharpCompiation(csSource As String, Optional additionalReferences As MetadataReference() = Nothing, Optional targetFramework As TargetFramework = TargetFramework.NetStandardLatest) As CSharp.CSharpCompilation

typo: Compilation

---
Refers to: src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/DefaultInterfaceImplementationTests.vb:18 in 24234ef. [](commit_id = 24234ef8bf6387e43fb5a0ac98fa0f8a070af98b, deletion_comment = False)

**Comment by jcouv** [comment]
>             Dim csCompiation = GetCSharpCompiation(csSource).EmitToImageReference()

typo: csCompilation

---
Refers to: src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/DefaultInterfaceImplementationTests.vb:43 in 24234ef. [](commit_id = 24234ef8bf6387e43fb5a0ac98fa0f8a070af98b, deletion_comment = False)

File: src/Compilers/VisualBasic/Test/Symbol/Microsoft.CodeAnalysis.VisualBasic.Symbol.UnitTests.vbproj (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/VisualBasic/Test/Symbol/Microsoft.CodeAnalysis.VisualBasic.Symbol.UnitTests.vbproj]: > What values would $(RoslynPortableTargetFrameworks) produce? (ie. what's the difference?)

I think it is: "net472;netcoreapp2.1"

---
In reply to: [288298770](https://github.com/dotnet/roslyn/pull/35957#discussion_r288298770) [](ancestors = 288298770)

File: src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs (hunk 2 above)
Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: > Not related to this PR: have we thought about the use of Obsolete attribute in DIM? Consider adding some tests if interesting.

Unless you have something specific in mind, I do not believe there is specific interaction between DIM and Obsolete attribute. 

---
In reply to: [288299204](https://github.com/dotnet/roslyn/pull/35957#discussion_r288299204) [](ancestors = 288299204)

**Comment by jcouv** [comment]
>             CompileAndVerify(comp1, expectedOutput:=If(ExecutionConditionUtil.IsMonoOrCoreClr, "C.M1", Nothing), verify:=VerifyOnMonoOrCoreClr)

I didn't quite follow this. 
The `verify` parameter controls whether to run PEVerify. If we can't run PEVerify on mono or coreCLR, it feels that should be handled inside our test framework, rather than in individual tests.

---
Refers to: src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/DefaultInterfaceImplementationTests.vb:98 in 24234ef. [](commit_id = 24234ef8bf6387e43fb5a0ac98fa0f8a070af98b, deletion_comment = False)

**Comment by AlekseyTs** [comment]
>             CompileAndVerify(comp1, expectedOutput:=If(ExecutionConditionUtil.IsMonoOrCoreClr, "C.M1", Nothing), verify:=VerifyOnMonoOrCoreClr)

> If we can't run PEVerify on mono or coreCLR, it feels that should be handled inside our test framework, rather than in individual tests.

It is the opposite, we cannot verify this code when the test is running on Desktop.

---
In reply to: [496691294](https://github.com/dotnet/roslyn/pull/35957#issuecomment-496691294) [](ancestors = 496691294)

---
Refers to: src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/DefaultInterfaceImplementationTests.vb:98 in 24234ef. [](commit_id = 24234ef8bf6387e43fb5a0ac98fa0f8a070af98b, deletion_comment = False)

**Pull request merged**